DependencyInjection/Configuration.php

```diff
@@ -796,6 +797,10 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->ifTrue(function ($v) { return 'server_log' === $v['type'] && empty($v['host']); })
                 ->thenInvalid('The host has to be specified to use a ServerLogHandler')
             ->end()
+            ->validate()
+                ->ifTrue(function ($v) { return 'console' !== $v['type'] && $v['interactive_only']; })
+                ->thenInvalid('The interactive_only flag is only available with the console handler')
```

Comment:
We generally don't throw errors for flags that are ignored for other types, as this makes it a lot harder to override the type of handler used for another environment (we would have to explicitly reset other fields).

All our validation rules are about settings that are required for a given type.

Reply:
That means you can set any setting for any type though, but OK I guess, I'll remove it.